src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,6 +1596,1436 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
+            }
+        }
+
+        if (!UpdateVictim())
+            return;
+
+        DoMeleeAttackIfReady();
+    }
+
+private:
+    EventMap _events;
+    ObjectGuid _playerGUID;
+};
+
+enum SkitterwebMatriarch
+{
+    NPC_SKITTERWEB_STALKER                      = 44908,
+
+    SPELL_SKITTERWEB                            = 83827,
+    SPELL_SUMMNON_SPIDERLINGS                   = 87084,
+    SPELL_VENOM_SPLASH                          = 79607,
+
+    EVENT_MATRIARCH_AGGRO                       = 1,
+    EVENT_SUMMON_SPIDERLINGS                    = 4,
+    EVENT_VENOM_SPLASH                          = 5,
+
+    ANIMKIT_MATRIARCH_INTERACT                  = 1,
+    ANIMKIT_MATRIARCH_POSITION1                 = 865,
+    ANIMKIT_MATRIARCH_POSITION2                 = 866
+};
+
+// 44906 - Skitterweb Matriarch
+struct npc_silverpine_skitterweb_matriarch : public ScriptedAI
+{
+    npc_silverpine_skitterweb_matriarch(Creature* creature) : ScriptedAI(creature), _alreadyPulled(false) { }
+
+    void JustAppeared() override
+    {
+        _alreadyPulled = false;
+
+        me->SetDisableGravity(true);
+        me->SetHover(true);
+
+        me->SetAIAnimKitId(ANIMKIT_MATRIARCH_POSITION1);
+
+        me->SetReactState(REACT_PASSIVE);
+
+        me->CastSpell(nullptr, SPELL_SKITTERWEB, true);
+    }
+
+    void Reset() override
+    {
+        _events.Reset();
+    }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        if (!_alreadyPulled)
+        {
+            _alreadyPulled = true;
+
+            me->SetAIAnimKitId(ANIMKIT_RESET);
+            me->PlayOneShotAnimKitId(ANIMKIT_MATRIARCH_POSITION2);
+
+            _events.ScheduleEvent(EVENT_MATRIARCH_AGGRO, 2s + 500ms);
+        }
+        else
+        {
+            _events.ScheduleEvent(EVENT_SUMMON_SPIDERLINGS, 2s);
+            _events.ScheduleEvent(EVENT_VENOM_SPLASH, 4s, 7s);
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_MATRIARCH_AGGRO:
+                    me->SetDisableGravity(false);
+                    me->SetHover(false);
+                    me->GetMotionMaster()->MoveFall();
+                    _events.ScheduleEvent(EVENT_MATRIARCH_AGGRO + 1, 1s);
+                    break;
+
+                case EVENT_MATRIARCH_AGGRO + 1:
+                    for (Creature* stalker : _stalkerList)
+                        stalker->RemoveAura(SPELL_SKITTERWEB);
+                    me->SetAIAnimKitId(ANIMKIT_MATRIARCH_INTERACT);
+                    me->SetHomePosition(me->GetPosition());
+                    _events.ScheduleEvent(EVENT_MATRIARCH_AGGRO + 2, 1s);
+                    break;
+
+                case EVENT_MATRIARCH_AGGRO + 2:
+                    me->SetReactState(REACT_AGGRESSIVE);
+                    _events.ScheduleEvent(EVENT_SUMMON_SPIDERLINGS, 2s);
```

Comment:
you could move the scheduling of `EVENT_SUMMON_SPIDERLINGS` and `EVENT_VENOM_SPLASH` into a new method and just call it here and in `JustEngagedWith` to have the scheduling of it in one place

Reply:
Feedback applied.